apps/src/dance/ai/DanceAiModalHeader.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {useMemo} from 'react';
+import moduleStyles from './dance-ai-modal.module.scss';
+import aiBotMini from '@cdo/static/dance/ai/bot/ai-bot-mini.svg';
+const i18n = require('../locale');
+
+import {DanceAiModelItem} from './types';
+
+import EmojiIcon from './EmojiIcon';
+
+interface DanceAiModalHeaderProps {
+  onClose: () => void;
+  handleStartOverClick: () => void;
+  selectedEmojis: DanceAiModelItem[];
+  slotCount: number;
+}
+
+const DanceAiModalHeader: React.FunctionComponent<DanceAiModalHeaderProps> = ({
+  onClose,
+  handleStartOverClick,
+  selectedEmojis,
+  slotCount,
+}) => {
+  const headerContent = useMemo(() => {
+    // The header comes from a localized string like this: "generate {input} effect".
+    // We split the localized string on the "{input}", and rebuild the HTML but with
+    // the emoji box in place of that "{input}".
+    const INPUT_KEY = '__CDO_INPUTS__';
+    const headerTextSplit = i18n
+      .danceAiModalHeading({input: INPUT_KEY})
+      .split(INPUT_KEY);
+
+    const filledSelectedEmojis: (DanceAiModelItem | undefined)[] = [
+      ...selectedEmojis,
+    ];
+    filledSelectedEmojis.length = slotCount;
+    filledSelectedEmojis.fill(undefined, selectedEmojis.length, slotCount);
+
+    return (
+      <>
+        {headerTextSplit[0]}
+        <div
```

Comment:
We could even pull out the three emoji input container into a separate component/`useMemo` since the `headerTextSplit` should stay constant?